r_reshape_long_wide/r_reshape_long_wide.md

```diff
@@ -0,0 +1,668 @@
+
+All of the other columns headers (the cleaned up version, without suffixes) will be the contents of our column that holds variable names, and the data within cells will be placed in the column that holds values.
+
+The final data will therefore look something like this (just a few rows shown):
+
+| biosample_id | key |	value |
+| --------- | --- | --- |
+| 3380629109 | subject_id | 876123 |
+| 3380629109 | collection_date | 2019-05-14 |
+| 3380629109 | collection_time | 09:15:00 |
+| 3380629109 | sample_type | blood |
+| 3380629109 | collection_method | venipuncture |
+| 3380629109 | collected\_by\_id | 4511 |
+
+Without having written any code, we've gotten an idea of what we want to accomplish:
+
+* split the data frame into two events
+* get rid of suffixes in column names
+* pivot all the column data except for `biosample_id` into a long format
+
+Let's jump into R and get started.  In `reshape_data_exercises.Rmd` in your RStudio environment, start by loading this data and taking a look at it (look at and run the code in lines 150-165).
+
+### Dealing With Suffixes
+
+All done bringing in the data and looking at it?  Great!  Time to move along to data reshaping.
+
+Our first challenge is to get all the `_1` variables, along with `subject_id` into one object, and all the `_2` variables, along with `subject_id`, into another.  We can do this using `dplyr`.
+
+Read and execute the code in lines 170-225 to create `biosample_first` and `biosample_second`, which will capture the two sets of data neatly (and even add an additional field giving the 1 or 2 sequence number, in case that ends up being important).  
+
+We'll use some commands that you already know, like `select`, as well as some that you might not be as familiar with, like `rename_with`.  Now that you've gotten some experience using help files, try using the `?` functionality and trying out `?rename_with` (or other function names) in the Console.
+
+### Pivoting Wide to Long
+
+Do you now have two objects, called `biosample_first` and `biosample_second`?  Great!
+
+At this point, you might be thinking, "Hey!  If we just stack `biosample_first` and `biosample_second`, we'll be done, and have some tidy data!".  You're not wrong, and you might be tempted to do just that and not bother with transforming wide data to long data.  Just remember that once you get data into a long format, it's very versatile, and it's worth the effort to put your data into a long format first.
+
+Go to lines 230 - 265 and read through the explanations and run the code in this section of the file.
+
+By the end of this section, you should have a `biosample_data_long` object that has our data in long format.
+
+### Pivoting Long to Wide
+
+By now, you should have long data that's ready to be pivoted to wide -- but this time, it will be wide data that's tidy, a vast improvement over the data we began with.
```

Comment:
I recommend moving the "we would like data that looks like this" table from the rmd file to this file, because reading the raw md table format will be hard for folks that aren't used to it. Here it will render as a nice table.

Reply:
Good call good call!